src/test/java/FlowTest.java

```diff
@@ -0,0 +1,121 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://github.com/teragrep/teragrep/blob/main/LICENSE>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+import com.teragrep.flow.FkFlowRequest;
+import com.teragrep.flow.Flow;
+import com.teragrep.flow.FlowResponse;
+import com.teragrep.flow.PartialFlowResponse;
+import jakarta.json.Json;
+import jakarta.json.JsonArray;
+import jakarta.json.JsonArrayBuilder;
+import jakarta.json.JsonObjectBuilder;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FlowTest {
+
+    @Test
+    public void flowRequestTest()  {
+        // Create fake request
+        Flow fakeFlow = new FkFlowRequest();
+
+        // Build data for expected response
+        JsonArrayBuilder arrayBuilder = Json.createArrayBuilder();
+        arrayBuilder.add(Json.createObjectBuilder().add("id",1).add("name","flow1"))
+                .add(Json.createObjectBuilder().add("id",2).add("name","flow2"));
+        JsonArray flowsArray = arrayBuilder.build();
+
+        FlowResponse fakeFlowResponse = Assertions.assertDoesNotThrow(fakeFlow::flowResponse);
+        FlowResponse realFlowResponse = new FlowResponse(flowsArray);
+
+        // Assertions
+        Assertions.assertFalse(flowsArray.isEmpty());
+        // Asserting that contents equal to each other
+        Assertions.assertEquals(realFlowResponse.toString(), fakeFlowResponse.toString());
```

Comment:
equality between objects must be tested with implementation of equals and hashCode methods, not via String equality